src/zcl_abapgit_environment.clas.abap

```diff
@@ -1,82 +1,83 @@
 CLASS zcl_abapgit_environment DEFINITION
   PUBLIC
-  CREATE PUBLIC .
-
+  FINAL
+  CREATE PRIVATE
+  GLOBAL FRIENDS zcl_abapgit_factory.
   PUBLIC SECTION.
-
-    CLASS-METHODS is_sap_cloud_platform
-      RETURNING
-        VALUE(rv_cloud) TYPE abap_bool.
-
-    CLASS-METHODS is_merged
-      RETURNING
-        VALUE(rv_is_merged) TYPE abap_bool.
-
-    CLASS-METHODS is_repo_object_changes_allowed
-      RETURNING VALUE(rv_allowed) TYPE abap_bool.
+    INTERFACES zif_abapgit_environment.
   PROTECTED SECTION.
-
-    CLASS-DATA gv_cloud TYPE abap_bool VALUE abap_undefined ##NO_TEXT.
-    CLASS-DATA gv_is_merged TYPE abap_bool VALUE abap_undefined ##NO_TEXT.
-    CLASS-DATA gv_client_modifiable TYPE abap_bool VALUE abap_undefined.
   PRIVATE SECTION.
+    DATA mv_cloud TYPE abap_bool VALUE abap_undefined ##NO_TEXT.
+    DATA mv_is_merged TYPE abap_bool VALUE abap_undefined ##NO_TEXT.
+    DATA mv_client_modifiable TYPE abap_bool VALUE abap_undefined ##NO_TEXT.
 ENDCLASS.
 
 
 
 CLASS ZCL_ABAPGIT_ENVIRONMENT IMPLEMENTATION.
 
 
-  METHOD is_merged.
+  METHOD zif_abapgit_environment~compare_with_inactive.
+    rv_result = zif_abapgit_environment~is_sap_cloud_platform( ).
+  ENDMETHOD.
+
 
+  METHOD zif_abapgit_environment~is_merged.
     DATA lo_marker TYPE REF TO data ##NEEDED.
 
-    IF gv_is_merged = abap_undefined.
+    IF mv_is_merged = abap_undefined.
       TRY.
           CREATE DATA lo_marker TYPE REF TO ('LIF_ABAPMERGE_MARKER')  ##no_text.
           "No exception --> marker found
-          gv_is_merged = abap_true.
+          mv_is_merged = abap_true.
 
         CATCH cx_sy_create_data_error.
-          gv_is_merged = abap_false.
+          mv_is_merged = abap_false.
       ENDTRY.
     ENDIF.
-
-    rv_is_merged = gv_is_merged.
-
+    rv_result = mv_is_merged.
   ENDMETHOD.
 
 
-  METHOD is_repo_object_changes_allowed.
+  METHOD zif_abapgit_environment~is_repo_object_changes_allowed.
     DATA lv_ind TYPE t000-ccnocliind.
 
-    IF gv_client_modifiable = abap_undefined.
+    IF mv_client_modifiable = abap_undefined.
       SELECT SINGLE ccnocliind FROM t000 INTO lv_ind
              WHERE mandt = sy-mandt.
       IF sy-subrc = 0
           AND ( lv_ind = ' ' OR lv_ind = '1' ). "check changes allowed
-        gv_client_modifiable = abap_true.
+        mv_client_modifiable = abap_true.
       ELSE.
-        gv_client_modifiable = abap_false.
+        mv_client_modifiable = abap_false.
       ENDIF.
     ENDIF.
-    rv_allowed = gv_client_modifiable.
+    rv_result = mv_client_modifiable.
   ENDMETHOD.
 
 
-  METHOD is_sap_cloud_platform.
+  METHOD zif_abapgit_environment~is_restart_required.
+    rv_result = abap_false.
```

Comment:
please add comment describing where this is used, currently it risks being deleted

Reply:
comment is added